src/System.Device.Gpio/System/Device/Pwm/PwmChannel.cs

```diff
@@ -79,6 +87,23 @@ protected virtual void Dispose(bool disposing)
             }
         }
 
+        private static bool IsBeagleBoneKernel()
+        {
+            try
+            {
+                string kernelVersionInfo = IO.File.ReadAllText(@"/proc/version");
+                // The String.Contains(string, StringComparison) overload wasn't introduced until .Net Standard 2.1
+                return kernelVersionInfo.IndexOf("beagle", StringComparison.InvariantCulture) > 0;
+            }
+            catch (SystemException ex) when (ex is IO.IOException || ex is Security.SecurityException || ex is UnauthorizedAccessException)
```

Comment:
One more comment here, can we make t his catch all exceptions so that we alwasy just fall back to old behavior in case of anything going wrong with the new codepath?

Reply:
Done @joperezr. I think that's addresses all of your feedback.
Thanks for your patience and working through this with me.

Reply:
Thank you for working through and for the great contribution!